### DISCUSSION TITLE
HY000 - unavailable: unexpected EOF

### DISCUSSION THREAD

**Discussion by RohanKapurDEV** [planetscale/discussion#342] [Q&A]
I have the following code in rust that errors out:

```rust
#[tokio::main]
async fn main() {
    dotenv().ok();
    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");

    let conn_pool = MySqlPoolOptions::new()
        .max_connections(5)
        .connect(&database_url)
        .await
        .expect("Failed to create pool");

    let mut rows = sqlx::query("SELECT * FROM solend_obligation").fetch(&conn_pool);

    while let Some(row) = rows.try_next().await.unwrap() {
        // ... some work
    }
}

```

The error happens on the `rows.try_next().await.unwrap()` bit: `thread 'main' panicked at 'called Result::unwrap() on an Err value: Database(MySqlDatabaseError { code: Some("HY000"), number: 1105, message: "unavailable: unexpected EOF" })', src/main.rs:29:49`.

Not sure whats going on since this was working just fine last night. Can't seem to find anything in the docs. Any help would be appreciated

**Comment by fooforge** [comment]
Hi @RohanKapurDEV,

I have taken a look at your database's metrics and I'm not seeing a drastic increased error rate whatsoever. Is your application erroring out immediately or is it hanging for a while before it throws an error? I'm asking because what I see is a single instance within the past 24 hours for where a transaction has been killed at the 20 seconds mark, and I suspect it's due to your application keeping the transaction open for too long.

Transactions on PlanetScale need to complete within 20 seconds. This is a deliberate hard limit that exists for performance reasons.

It can be worked around temporarily by switching from the default OLTP to the OLAP workload mode, but it's generally not advised as it increases the chance to hit other query limits. You can switch to OLAP on a per-session basis by issuing a `set workload='olap';` before running any transactions, but it cannot be enabled globally. The best solution still is is to design your application and queries in a way so that transactions complete within the 20 seconds time window.
